src/Factory.php

```diff
@@ -154,7 +155,12 @@ public function createInput($inputSpecification)
             ));
         }
 
-        $input = $managerInstance ?: new $class();
+        if ($class === Input::class || is_subclass_of($class, Input::class)) {
```

Comment:
`is_a` seems can be used, see https://3v4l.org/kjoXX#v8.4.10

```suggestion
        if (is_a($class, Input::class, true)) {
```

Reply:
Thanks 😄